lib/install/validate/validator_test.go

```diff
@@ -117,6 +119,72 @@ func TestParseURL(t *testing.T) {
 	}
 }
 
+func TestDenormalizeRegistryURLs(t *testing.T) {
+	parseEntry := func(input string) (outputs []string) {
+		rs := registry.ParseEntry(input)
+		r, ok := rs.(registry.URLEntry)
+		if !ok {
+			t.FailNow()
+		}
+
+		var err error
+		entries, err := denormalizeRegistryURL(r)
+		if err != nil {
+			t.FailNow()
+		}
+		for _, entry := range entries {
+			outputs = append(outputs, entry.String())
+		}
+
+		return
+	}
+
+	testCases := map[string][]string{
+		"foo.local": {
```

Comment:
So this still has me concerned... this is essentially saying that we expect if someone supplies foo.local then we assume and only assume https?  If a user puts in foo.local, we can't really assume anything right?  We need to support http and https if a user supplies only a fqdn without a scheme or port right?  Should we not try all combinations?